src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -153,94 +208,100 @@ def __str__(self):
-    'AzureUSGovernment',
-    endpoints=CloudEndpoints(
-        management='https://management.core.usgovcloudapi.net/',
-        resource_manager='https://management.usgovcloudapi.net/',
-        sql_management='https://management.core.usgovcloudapi.net:8443/',
-        batch_resource_id='https://batch.core.usgovcloudapi.net/',
-        gallery='https://gallery.usgovcloudapi.net/',
-        active_directory='https://login.microsoftonline.us',
-        active_directory_resource_id='https://management.core.usgovcloudapi.net/',
-        active_directory_graph_resource_id='https://graph.windows.net/',
-        microsoft_graph_resource_id='https://graph.microsoft.us/',
-        vm_image_alias_doc='https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json',  # pylint: disable=line-too-long
-        media_resource_id='https://rest.media.usgovcloudapi.net',
-        ossrdbms_resource_id='https://ossrdbms-aad.database.usgovcloudapi.net'),
-    suffixes=CloudSuffixes(
-        storage_endpoint='core.usgovcloudapi.net',
-        keyvault_dns='.vault.usgovcloudapi.net',
-        sql_server_hostname='.database.usgovcloudapi.net',
-        acr_login_server_endpoint='.azurecr.us'))
-
-AZURE_GERMAN_CLOUD = Cloud(
-    'AzureGermanCloud',
-    endpoints=CloudEndpoints(
-        management='https://management.core.cloudapi.de/',
-        resource_manager='https://management.microsoftazure.de',
-        sql_management='https://management.core.cloudapi.de:8443/',
-        batch_resource_id='https://batch.cloudapi.de/',
-        gallery='https://gallery.cloudapi.de/',
-        active_directory='https://login.microsoftonline.de',
-        active_directory_resource_id='https://management.core.cloudapi.de/',
-        active_directory_graph_resource_id='https://graph.cloudapi.de/',
-        microsoft_graph_resource_id='https://graph.microsoft.de',
-        vm_image_alias_doc='https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json',  # pylint: disable=line-too-long
-        media_resource_id='https://rest.media.cloudapi.de',
-        ossrdbms_resource_id='https://ossrdbms-aad.database.cloudapi.de'),
-    suffixes=CloudSuffixes(
-        storage_endpoint='core.cloudapi.de',
-        keyvault_dns='.vault.microsoftazure.de',
-        sql_server_hostname='.database.cloudapi.de'))
-
-
-KNOWN_CLOUDS = [AZURE_PUBLIC_CLOUD, AZURE_CHINA_CLOUD, AZURE_US_GOV_CLOUD, AZURE_GERMAN_CLOUD]
+try:
+    arm_cloud_dict = json.loads(urlretrieve(os.getenv('ARM_CLOUD_METADATA_URL')))
+    cli_cloud_dict = convert_arm_to_cli(arm_cloud_dict)
+    if 'AzureCloud' in cli_cloud_dict:
+        cli_cloud_dict['AzureCloud'].endpoints.active_directory = 'https://login.microsoftonline.com'  # pylint: disable=line-too-long # change once active_directory is fixed in ARM for the public cloud
+    KNOWN_CLOUDS = list(cli_cloud_dict.values())
+except Exception:  # pylint: disable=broad-except
```

Comment:
I will suggest we remove the try~catch and keep the hard coded ones; then at the end, detect the env var is set, then load from there.

Reply:
makes sense, done